### PR TITLE
Remove scss from frontend

### PR DESCRIPTION
**Pull request by lewisl9029**
## Summary

This PR is a first pass at migrating frontend from SCSS modules to plain CSS modules with native nesting syntax. This allows us to remove all the generated files in __generated/scss, and reduce the size of __generated/index.css by about 70%, leaving only tailwind (will be working on getting rid of this next) and vanilla extract styles, relying instead on the newly added first-class CSS modules support in Reflame.

Native nesting syntax is _almost_ supported in all major evergreen browsers, with the only exception being Firefox, which has just [enabled it in Nightly](https://bugzilla.mozilla.org/show_bug.cgi?id=1835066), so we'll be transpiling it away in both Vite and Reflame for now, but can revisit at a later time to enable smaller CSS output.

Basically, the only major difference between nesting in SCSS and nesting in native CSS is the requirement for the nesting symbol `&` to be present before every nested selector, whereas in SCSS it could be omitted for nested class and element selectors. Some examples to illustrate:

Valid SCSS nesting, invalid CSS nesting:
```
.blah {
  .nested {}
}
```

Valid CSS nesting:
```
.blah {
  & .nested {}
}
```

I actually ended up doing most of the migration manually before I realized I could just do a search and replace for `	.`, `	>`, `	span`, etc to add & in front of each. 🤦 

~~Note this PR is not ready to be merged yet due to some issues I've discovered with Reflame's css import ordering (hence the draft state). Asides from that the output seems to be identical to the current Reflame preview in main. You can verify for yourself using the latest preview [here](https://preview.highlight.io/?%7Er_preview=%7B%22action%22%3A%22start%22%2C%22variantId%22%3A%2201H2W9CD0ESX3QJ1QYN1C1YGGK%22%2C%22region%22%3A%22sjc%22%2C%22variantData%22%3A%22%7B%5C%22type%5C%22%3A%5C%22branch%5C%22%2C%5C%22branch%5C%22%3A%5C%22remove-scss-from-frontend%5C%22%2C%5C%22forkSpecifier%5C%22%3A%5C%22lewisl9029%2Fhighlight%5C%22%2C%5C%22githubOwnerName%5C%22%3A%5C%22highlight%5C%22%2C%5C%22githubRepositoryName%5C%22%3A%5C%22highlight%5C%22%7D%22%7D).~~ The import ordering issue has been fixed! PR is now ready for code review and testing. 

Would probably recommend reviewing as a whole PR from the [files changed tab](https://github.com/highlight/highlight/pull/5642/files) instead of commit by commit, and skipping over most of the CSS changes as they're all fairly mechanical syntax replacements (I've left a comment at the only notable exception [here](https://github.com/highlight/highlight/pull/5642#discussion_r1229094950)). Testing the actual app in Render and [Reflame](https://preview.highlight.io/?%7Er_preview=%7B%22action%22%3A%22start%22%2C%22variantId%22%3A%2201H2W9CD0ESX3QJ1QYN1C1YGGK%22%2C%22region%22%3A%22sjc%22%2C%22variantData%22%3A%22%7B%5C%22type%5C%22%3A%5C%22branch%5C%22%2C%5C%22branch%5C%22%3A%5C%22remove-scss-from-frontend%5C%22%2C%5C%22forkSpecifier%5C%22%3A%5C%22lewisl9029%2Fhighlight%5C%22%2C%5C%22githubOwnerName%5C%22%3A%5C%22highlight%5C%22%2C%5C%22githubRepositoryName%5C%22%3A%5C%22highlight%5C%22%7D%22%7D) previews is probably going to be a better use of reviewer time. 

## How did you test this change?

~~Poking around in the local Vite build and Reflame preview, everything seems good outside of the known CSS import ordering issue that seems to only be affecting the rrweb player layout:~~

Screenshot 2023-06-13 at 11 51 29 PM

~~I'll be working on a fix for this tomorrow, but the fix is purely on the Reflame side, and shouldn't require any changes in this PR.~~ Update: This is done!

Everything seems to look good on my side from the Reflame preview and local Vite build.

## Are there any deployment considerations?

Would appreciate a Render preview to do a sanity check on the production Vite output. 🙏

### CODE REVIEW

File: frontend/src/pages/ErrorsV2/ErrorStackTrace/ErrorStackTrace.module.css [hunk 1]
```diff
@@ -57,13 +57,15 @@
 .collapsibleWrapper {
 	width: 100%;
 
-	&#{&} {
-		h2 {
-			font-size: 14px !important;
-			font-weight: 400;
-		}
+	/* Not quite sure what the original selector here was doing, replacing with simple nesting for now? */
+	/* &#{&} { */
```
Comment (lewisl9029) [inline frontend/src/pages/ErrorsV2/ErrorStackTrace/ErrorStackTrace.module.css:61]: As mentioned in the comment, my SCSS-fu is pretty rusty and I couldn't figure out what this selector was doing, so this was my probably incorrect attempt at replicating it. 

But on a closer look, I also couldn't find any references to `collapsibleWrapper` anywhere, so maybe this entire class is safe to remove?

**Review by mayberryzane — APPROVED** [review]
looks good in preview env!

**Comment by Vadman97** [comment]
looks good! didn't have a way to deploy this fork as a PR, but i accidentally deployed to render prod and it looks good
image

**Comment by lewisl9029** [comment]
@Vadman97 sorry, I could swear there were no conflicts when I asked for the review. They're resolved now, Reflame preview still looks good as far as I can see. 🙏

**Comment by Vadman97** [comment]
looks like the new `render:test` doesn't work in forks because it requires s3 creds. fixing that

**Comment by Vadman97** [comment]
hey @lewisl9029 , [fix](https://github.com/highlight/highlight/blob/fac7f5097313d914c2c9853b1a4ab2956b7df228/.github/workflows/turbo.yml#L74-L76) is now in main. mind rebasing?

**Comment by lewisl9029** [comment]
@Vadman97 thanks for the quick fix! Updated! :)

**Comment by lewisl9029** [comment]
@Vadman97 looks like the vercel deploy failed, but I can't see the logs. Any ideas what's going on?

**Pull request merged**